reference.go

```diff
@@ -1,6 +1,9 @@
 // Package reference provides a general type to represent any way of referencing images within the registry.
 // Its main purpose is to abstract tags and digests (content-addressable hash).
 //
+// Note: Hash implementation packages must be imported in order
+// for digest parsing to work. Example package for sha256: [crypto/sha256]
+//
```

Comment:
Perhaps we should make this a separate section below the Grammar; we could borrow some of the wording from https://pkg.go.dev/github.com/opencontainers/go-digest#readme-usage

Might also be good to include this information in GoDoc for individual functions to describe their behavior.